csrc/fusion.cpp

```diff
@@ -794,23 +794,32 @@ void Fusion::aliasOutputToInput(
     if (input->getDataType().value() != output->getDataType().value()) {
       output = castOp(input->getDataType().value(), output);
     }
-  }
 
-  NVF_ERROR(
-      isAliasCompatible(input, output),
-      "The input and output values are not alias-compatible.");
-  // Let integration hide any output that wasn't a fusion output when
-  // `aliasOutputToInput` was called. For example, running mean and var for
-  // batch norm.
-  io_alias_[output] = AliasInfo{
-      .type = type,
-      .aliased_io = input,
-      .hide_output = !output->isFusionOutput()};
-
-  // TODO: output should be marked at the end of fusion definition #1488
-  if (!output->isFusionOutput()) {
-    addOutput(output);
+    NVF_ERROR(
+        isAliasCompatible(input, output),
+        "The input and output values are not alias-compatible.");
+    // Let integration hide any output that wasn't a fusion output when
+    // `aliasOutputToInput` was called. For example, running mean and var for
+    // batch norm.
+    io_alias_[output] = AliasInfo{
+        .type = type,
+        .aliased_io = input,
+        .hide_output = !output->isFusionOutput()};
+
+    // TODO: output should be marked at the end of fusion definition #1488
+    if (!output->isFusionOutput()) {
+      addOutput(output);
+    }
+    return;
   }
+
+  // AllocationType::Evaluate
+  NVF_ERROR(type == AllocationType::Evaluate);
+  NVF_CHECK(
+      output->isFusionOutput(),
+      "Only fusion outputs can be expression evaluated.");
+  io_alias_[output] =
```

Comment:
Nit: I think code will be cleaner if it tries to handle Evaluate first. That is, instead of 

```
if (inplace-update) {
  long code
  return;
}
short code
```
do 
```
if (evaluate) {
  short code
  return;
}
long code
```

This way, readers quickly cross out one case after reading "short code" instead of "long code".